packages/flutter/test/cupertino/sheet_test.dart

```diff
@@ -649,6 +649,99 @@ void main() {
     expect(find.text('Page: /next'), findsOneWidget);
   });
 
+  testWidgets('content does not go below the bottom of the screen', (WidgetTester tester) async {
+    final GlobalKey scaffoldKey = GlobalKey();
+
+    await tester.pumpWidget(
+      CupertinoApp(
+        home: CupertinoPageScaffold(
+          key: scaffoldKey,
+          child: Center(
+            child: Column(
+              children: <Widget>[
+                const Text('Page 1'),
+                CupertinoButton(
+                  onPressed: () {
+                    Navigator.push<void>(
+                      scaffoldKey.currentContext!,
+                      CupertinoSheetRoute<void>(
+                        builder: (BuildContext context) {
+                          return CupertinoPageScaffold(child: Container());
+                        },
+                      ),
+                    );
+                  },
+                  child: const Text('Push Page 2'),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    await tester.tap(find.text('Push Page 2'));
+    await tester.pumpAndSettle();
+
+    expect(tester.getSize(find.byType(Container)).height, 600.0 - (600.0 * 0.08));
```

Comment:
Nit: Maybe declare a _kTopGapRatio variable in the test and use it here, just so it's more obvious where the 0.08 comes from.